x-pack/plugins/data_enhanced/server/search/session/monitoring_task.ts

```diff
@@ -39,6 +43,8 @@ function searchSessionRunner(
           logger.debug('Search sessions are disabled. Skipping task.');
           return;
         }
+        if (aborted$.getValue()) return;
```

Comment:
This looks OK, but it's worth clarifying that if (for some reason) you return here **before** Task Manager actually calls _cancel_, then you'll be wiping out your State as TM will think you've just completed the task without returning any state.

It doesn't look like you're using task state, so I think this is fine, but I felt it was worth flagging

Reply:
Thanks for the flag!